service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -229,6 +234,18 @@ public DatasetPreviewTablesResponse listDatasetPreviewTables(
     return new DatasetPreviewTablesResponse().tables(tableMetadataList);
   }
 
+  private void validateSchema(String metaData) {
+    JsonNode node;
+    try {
+      node = objectMapper.readValue(metaData, JsonNode.class);
+    } catch (JsonProcessingException e) {
+      throw new IllegalMetadataException(e);
+    }
+    if (node.getNodeType() != JsonNodeType.OBJECT) {
+      throw new BadRequestException("catalogEntry/metadata must be a json object");
+    }
```

Comment:
I think it's better to initialize a variable at its declaration point, if possible. Moving the `if` inside the `try` would allow you do do that.
```suggestion
    try {
      JsonNode node = objectMapper.readValue(metaData, JsonNode.class);
      if (node.getNodeType() != JsonNodeType.OBJECT) {
        throw new BadRequestException("catalogEntry/metadata must be a json object");
      }
    } catch (JsonProcessingException e) {
      throw new IllegalMetadataException(e);
    }
```